theano/tensor/opt.py

```diff
@@ -5841,3 +5841,22 @@ def local_add_mul_fusion(node):
 def local_grad_clip(node):
     if isinstance(node.op, theano.gradient.GradClip):
         return node.inputs
+
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@gof.local_optimizer([T.alloc])
+def local_double_alloc(node):
+    # Alloc(Alloc(m, x, 1), x, y) -> Alloc(m, x, y)
+    if node.op != T.alloc:
+        return False
+    if not node.inputs[0].owner or node.inputs[0].owner.op != T.alloc:
+        return False
+    if len(node.inputs) != 3 or len(node.inputs[0].owner.inputs) != 3:
```

Comment:
The opt should not be specific to only matrix. Make it more general for any ndim and there can be one or more 1 and it can be at any dimensions.